capsules/src/virtual_digest.rs

```diff
@@ -0,0 +1,136 @@
+//! Virtualize the Digest interface to enable multiple users of an underlying
+//! Digest hardware peripheral.
+
+use core::cell::Cell;
+use core::marker::PhantomData;
+use kernel::common::cells::OptionalCell;
+use kernel::common::leasable_buffer::LeasableBuffer;
+use kernel::common::{ListLink, ListNode};
+use kernel::hil::digest;
+use kernel::hil::digest::DigestType;
+use kernel::ReturnCode;
+
+pub struct VirtualMuxDigest<
+    'a,
+    A: kernel::hil::digest::HMACSha256 + digest::Digest<'a, T>,
```

Comment:
```suggestion
    A: digest::Digest<'a, T>,
```

If this virtualizes only on Digest, it shouldn't assume the Digest hardware supports a particular algorithm, correct?